actors/builtin/miner/miner_test.go

```diff
@@ -2260,24 +2230,59 @@ func (h *actorHarness) terminateSectors(rt *mock.Runtime, sectors *abi.BitField,
 		expectQueryNetworkInfo(rt, h)
 	}
 
-	{
-		// TODO minerstate
-		//rawPower, qaPower := miner.PowerForSectors(h.sectorSize, sectorInfos)
-		//rt.ExpectSend(builtin.StoragePowerActorAddr, builtin.MethodsPower.UpdateClaimedPower, &power.UpdateClaimedPowerParams{
-		//	RawByteDelta:         rawPower.Neg(),
-		//	QualityAdjustedDelta: qaPower.Neg(),
-		//}, abi.NewTokenAmount(0), nil, exitcode.Ok)
-	}
 	if big.Zero().LessThan(expectedFee) {
 		rt.ExpectSend(builtin.BurntFundsActorAddr, builtin.MethodSend, nil, expectedFee, nil, exitcode.Ok)
 		pledgeDelta := expectedFee.Neg()
 		rt.ExpectSend(builtin.StoragePowerActorAddr, builtin.MethodsPower.UpdatePledgeTotal, &pledgeDelta, big.Zero(), nil, exitcode.Ok)
 	}
+	// notify change to initial pledge
+	if len(sectorInfos) > 0 {
+		pledgeDelta := big.Zero()
+		for _, sector := range sectorInfos {
+			pledgeDelta = big.Add(pledgeDelta, sector.InitialPledge.Neg())
+		}
+		rt.ExpectSend(builtin.StoragePowerActorAddr, builtin.MethodsPower.UpdatePledgeTotal, &pledgeDelta, big.Zero(), nil, exitcode.Ok)
+	}
+	{
+		sectorPower := miner.PowerForSectors(h.sectorSize, sectorInfos)
+		rt.ExpectSend(builtin.StoragePowerActorAddr, builtin.MethodsPower.UpdateClaimedPower, &power.UpdateClaimedPowerParams{
+			RawByteDelta:         sectorPower.Raw.Neg(),
+			QualityAdjustedDelta: sectorPower.QA.Neg(),
+		}, abi.NewTokenAmount(0), nil, exitcode.Ok)
+	}
 
-	// TODO minerstate
-	//params := &miner.TerminateSectorsParams{Sectors: sectors}
-	//rt.Call(h.a.TerminateSectors, params)
-	//rt.Verify()
+	// create declarations
+	st := getState(rt)
+	deadlines, err := st.LoadDeadlines(rt.AdtStore())
+	require.NoError(h.t, err)
+
+	declarationMap := map[decKey]*miner.TerminationDeclaration{}
+	err = sectors.ForEach(func(id uint64) error {
+		dlIdx, pIdx, err := miner.FindSector(rt.AdtStore(), deadlines, abi.SectorNumber(id))
+		require.NoError(h.t, err)
+
+		declaration, ok := declarationMap[decKey{dlIdx, pIdx}]
+		if !ok {
+			declaration = &miner.TerminationDeclaration{
+				Deadline:  dlIdx,
+				Partition: pIdx,
+				Sectors:   bf(),
+			}
+			declarationMap[decKey{dlIdx, pIdx}] = declaration
+		}
+		declaration.Sectors.Set(id)
+		return nil
+	})
+	require.NoError(h.t, err)
+
+	declarations := []miner.TerminationDeclaration{}
+	for _, declaration := range declarationMap {
+		declarations = append(declarations, *declaration)
+	}
```

Comment:
The aggregation isn't strictly necessary here. It would be fine to submit a declaration per sector. Indeed that could be better to exercise the aggregation in the actor.